lib/vdsm/storage/storageServer.py

```diff
@@ -924,11 +924,22 @@ def createConnection(cls, conInfo):
 
 
 def connect(dom_type, con_defs):
+    if not con_defs:
+        log.warning("Connect: connection list is empty, dom_type=%s", dom_type)
```

Comment:
This is not the place to fix this. This is internal API, the caller must send
a non-emtpy list - if not this is a bug.

To make sure this never happens, just assert about the list:

   assert conn_defs